Kubescape panics when scanning a local Git branch without an upstream and `origin` fallback

# Describe the bug

Kubescape panics on scanning a local git branch when the branch does not have a remote set and a fallback remote named `origin` is missing.

# Environment

OS: Ubuntu 22.04 LTS
Version: git `master`, latest release

# Steps To Reproduce

Steps to reproduce the behavior:

1. Run the following commands:
```
git clone https://github.com/kubescape/kubscape
cd kubescape
make all
git checkout -b new-branch
git remote rename origin upstream
./kubescape scan examples/
```
2. Observe.

# Expected behavior

Kubescape reports that the origin is missing, and then performs a scan and reports its results as usual.

# Actual Behavior

Kubescape panics as follows:
```
$ ./kubescape scan examples/
[info] Kubescape scanner starting
[warning] unknown build number, this might affect your scan results. Please make sure you are updated to latest version
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x3195f6f]

goroutine 1 [running]:
github.com/kubescape/kubescape/v2/core/cautils.(*LocalGitRepository).GetRemoteUrl(0xc001825680)
        /home/user/kubescape/kubescape/core/cautils/localgitrepository.go:82 +0x1af
github.com/kubescape/kubescape/v2/core/cautils.metadataGitLocal({0x7fff899740ee?, 0x9?})
        /home/user/kubescape/kubescape/core/cautils/scaninfo.go:387 +0x90
github.com/kubescape/kubescape/v2/core/cautils.setContextMetadata(0xc001387540, {0x7fff899740ee, 0x9})
        /home/user/kubescape/kubescape/core/cautils/scaninfo.go:345 +0x37c
github.com/kubescape/kubescape/v2/core/cautils.scanInfoToScanMetadata(0xc0003d6b40)
        /home/user/kubescape/kubescape/core/cautils/scaninfo.go:276 +0x2d9
github.com/kubescape/kubescape/v2/core/cautils.NewOPASessionObj({0x0, 0x0, 0x0}, 0x0, 0xc0003d6b40)
        /home/user/kubescape/kubescape/core/cautils/datastructures.go:51 +0x85
github.com/kubescape/kubescape/v2/core/pkg/policyhandler.(*PolicyHandler).CollectResources(0xc000b1f7f0, {0xc0002d4e00, 0x6, 0x8}, 0xc0003d6b40)
        /home/user/kubescape/kubescape/core/pkg/policyhandler/handlenotification.go:37 +0x3f
github.com/kubescape/kubescape/v2/core/core.(*Kubescape).Scan(0x3c52d00?, 0xc0003d6b40)
        /home/user/kubescape/kubescape/core/core/scan.go:158 +0x6b9
github.com/kubescape/kubescape/v2/cmd/scan.getFrameworkCmd.func2(0x36cc8a0?, {0xc000479620, 0x2, 0x2})
        /home/user/kubescape/kubescape/cmd/scan/framework.go:106 +0x46c
github.com/kubescape/kubescape/v2/cmd/scan.GetScanCommand.func1(0xc0004d8600?, {0xc000429870, 0x1, 0x1?})
        /home/user/kubescape/kubescape/cmd/scan/scan.go:46 +0x166
github.com/spf13/cobra.(*Command).ValidateArgs(...)
        /home/vklokun/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1072
github.com/spf13/cobra.(*Command).execute(0xc0004d8600?, {0xc0004297d0?, 0x1?, 0x1?})
        /home/vklokun/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:885 +0x653
github.com/spf13/cobra.(*Command).ExecuteC(0xc0004d8300)
        /home/vklokun/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /home/vklokun/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
github.com/kubescape/kubescape/v2/cmd.Execute()
        /home/user/kubescape/kubescape/cmd/root.go:87 +0x2d
main.main()
        /home/user/kubescape/kubescape/main.go:9 +0x1d
```

# Additional context

This happens because we don’t check if a key is present when accessing the git config `Remotes` field in `localgitrepository.go:82`.

--- a/core/cautils/localgitrepository.go
+++ b/core/cautils/localgitrepository.go
@@ -79,10 +79,13 @@ func (g *LocalGitRepository) GetRemoteUrl() (string, error) {
 	}
 
 	const defaultRemoteName string = "origin"
-	if len(g.config.Remotes[defaultRemoteName].URLs) == 0 {
+	defaultRemote, ok := g.config.Remotes[defaultRemoteName]
+	if !ok {
+		return "", fmt.Errorf("did not find a default remote with name '%s'", defaultRemoteName)
+	} else if len(defaultRemote.URLs) == 0 {
 		return "", fmt.Errorf("expected to find URLs for remote '%s'", defaultRemoteName)
 	}
-	return g.config.Remotes[defaultRemoteName].URLs[0], nil
+	return defaultRemote.URLs[0], nil
 }
 
 // GetName get origin name without the .git suffix

--- a/core/cautils/localgitrepository_test.go
+++ b/core/cautils/localgitrepository_test.go
@@ -9,6 +9,8 @@ import (
 	"strings"
 	"testing"
 
+	configv5 "github.com/go-git/go-git/v5/config"
+	plumbingv5 "github.com/go-git/go-git/v5/plumbing"
 	"github.com/stretchr/testify/suite"
 )
 
@@ -172,4 +174,51 @@ func (s *LocalGitRepositoryTestSuite) TestGetFileLastCommit() {
 
 		}
 	})
+}
+
+func TestGetRemoteUrl(t *testing.T) {
+	testCases := []struct {
+		Name      string
+		LocalRepo LocalGitRepository
+		Want      string
+		WantErr   error
+	}{
+		{
+			Name: "Branch with missing upstream and missing 'origin' fallback should return an error",
+			LocalRepo: LocalGitRepository{
+				config: &configv5.Config{
+					Branches: make(map[string]*configv5.Branch),
+					Remotes:  make(map[string]*configv5.RemoteConfig),
+				},
+				head: plumbingv5.NewReferenceFromStrings("HEAD", "ref: refs/heads/v4"),
+			},
+			Want:    "",
+			WantErr: fmt.Errorf("did not find a default remote with name 'origin'"),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			localRepo := LocalGitRepository{
+				config: &configv5.Config{
+					Branches: make(map[string]*configv5.Branch),
+					Remotes:  make(map[string]*configv5.RemoteConfig),
+				},
+				head: plumbingv5.NewReferenceFromStrings("HEAD", "ref: refs/heads/v4"),
+			}
+
+			want := tc.Want
+			wantErr := tc.WantErr
+			got, gotErr := localRepo.GetRemoteUrl()
+
+			if got != want {
+				t.Errorf("Remote URLs don’t match: got '%s', want '%s'", got, want)
+			}
+
+			if gotErr.Error() != wantErr.Error() {
+				t.Errorf("Errors don’t match: got '%v', want '%v'", gotErr, wantErr)
+			}
+		},
+		)
+	}
 }

--- a/core/pkg/containerscan/gojayunmarshaller.go
+++ b/core/pkg/containerscan/gojayunmarshaller.go
@@ -64,7 +64,7 @@ func (pkgs *LinuxPkgs) UnmarshalJSONArray(dec *gojay.Decoder) error {
 	return nil
 }
 
-//--------Vul fixed in----------------------------------
+// --------Vul fixed in----------------------------------
 func (fx *FixedIn) UnmarshalJSONObject(dec *gojay.Decoder, key string) (err error) {
 
 	switch key {

--- a/core/pkg/containerscan/rawdatastrucutres.go
+++ b/core/pkg/containerscan/rawdatastrucutres.go
@@ -71,19 +71,19 @@ type PackageFile struct {
 
 // types to provide unmarshalling:
 
-//VulnerabilitiesList -s.e
+// VulnerabilitiesList -s.e
 type LayersList []ScanResultLayer
 
-//VulnerabilitiesList -s.e
+// VulnerabilitiesList -s.e
 type VulnerabilitiesList []Vulnerability
 
-//LinuxPkgs - slice of linux pkgs
+// LinuxPkgs - slice of linux pkgs
 type LinuxPkgs []LinuxPackage
 
-//VulFixes - information bout when/how this vul was fixed
+// VulFixes - information bout when/how this vul was fixed
 type VulFixes []FixedIn
 
-//PkgFiles - slice of files belong to specific pkg
+// PkgFiles - slice of files belong to specific pkg
 type PkgFiles []PackageFile
 
 func (v *ScanResultReport) AsFNVHash() string {

--- a/core/pkg/resourcehandler/remotegitutils.go
+++ b/core/pkg/resourcehandler/remotegitutils.go
@@ -40,7 +40,7 @@ func isGitTokenPresent(gitURL giturl.IGitAPI) bool {
 
 // Get the error message according to the provider
 func getProviderError(gitURL giturl.IGitAPI) error {
-	switch gitURL.GetProvider(){
+	switch gitURL.GetProvider() {
 	case "github":
 		return fmt.Errorf("%w", errors.New("GITHUB_TOKEN is not present"))
 	case "gitlab":

